### PR TITLE
add semantic-release and make the repo commitizen-friendly 

### LINKED ISSUE #37: Release latest progress
Assalam alykoum

It seems to me that the current version `4.0.1` is behind `master` branch. for example, the type definitions in the latest release don't include `polarCircleResolution`. and I noticed that the Quibla feature is not defined in the current type definitions (probably needs a separate issue. let me know and I will open a new one).

How are the releases triggered?

### PR DESCRIPTION
**Pull request by gimyboya**
- Added semantic-release to support semver and release to github and npm
- Added commitlint to lint commit message when created
- Added commitizen to help create commit messages with `commit` script
- Re-organized Nodjs ci github action into smaller jobs and added a release job.
- Remove the manual release github action.
- Added a contribution note and an explanation on the commit convention the repo follows.

**Before merging:** create a tag with the name `v4.0.3` at the same level where the current `4.0.3` tag exists
		  
**Note:** Once semantic-release is configured, any user with the permission to push commits to master or the branches listed on the .releaserc will be able to publish a release. It is recommended to protect those branches, for example with GitHub protected branches.

**After merging:** disable the `dry-run` for `semantic-release` in the Github action

close #37

### CODE REVIEW

**Comment by gimyboya** [comment]
This pr once merged should close #37

**Comment by gimyboya** [comment]
@z3bi @sgtsquiggs there is a need to greater a new tag with that name `v4.0.3` before merging this or so that `semantic-release` will start counting commits from the last release.

image

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -10,20 +10,80 @@ on:
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Testing on node ${{ matrix.node-version }} 🧪
+      uses: actions/setup-node@v1
+      with:
+          node-version: ${{ matrix.node-version }}
     - run: npm ci
-    - run: npm run build --if-present
     - run: npm test
+# # coverage uploading job
```
Comment (gimyboya) [inline .github/workflows/node.js.yml:48]: @z3bi @sgtsquiggs I also suggest to move code coverage to GitHub actions and eliminate the need for travis.

I have set up the job already if you guys want that just uncomment and add `CODECOV_TOKEN ` as well as adjust the directory (forgot to do that)

**Comment by z3bi** [comment]
Would the idea be to then commit to a `develop` branch and once it's time to release do a single PR from `develop` to `master`?

**Comment by gimyboya** [comment]
@z3bi yes exactly.

to quote the original git-flow article which can be found [here](https://nvie.com/posts/a-successful-git-branching-model/)

> The master branch at origin should be familiar to every Git user. Parallel to the master branch, another branch exists called develop.

> We consider origin/master to be the main branch where the source code of HEAD always reflects a production-ready state.

> We consider origin/develop to be the main branch where the source code of HEAD always reflects a state with the latest delivered development changes for the next release. Some would call this the “integration branch”. This is where any automatic nightly builds are built from.

using the git-flow branching model we obtain a more clear/organized way of collaboration

**Comment by sgtsquiggs** [comment]
@gimyboya is it that we needed to add 4.0.3 specifically, or do we just need a tag immediately before merging this?

**Comment by gimyboya** [comment]
@sgtsquiggs sorry I didn't explain the logic behind the creation of a new tag.

For `semantic-release` to be able to [start counting commits starting from the last release](https://semantic-release.gitbook.io/semantic-release/usage/configuration#existing-version-tags)
 of this repo (which at the time of this writing is `4.0.3` ) we need to create a tag that `semantic-release` recognizes for it to start analyzing commits from that tag and knowing that the last version was `v4.0.3` and the next one will be from there onwards otherwise, it will release the next version start from `v1.0.0`.

currently, this repo already has a version tag with the following format `4.0.3` at the last release commit, however, semantic release ignores that tag because it's looking for a tag with the following format `v${version}`.

it is possible to [provide a different tag format](https://semantic-release.gitbook.io/semantic-release/usage/configuration#tagformat) for `semantic-release` with the `-t` or `--tag-format` flags when using the CLI. unfortunately, since we are using the semantic-release from [`cycjimmy/semantic-release-action`](https://github.com/cycjimmy/semantic-release-action) GitHub action, I couldn't find a way to provide that option to `semantic-release` as it's not exposed in the GitHub action configuration.

Thus I suggested first create a `v4.0.3` tag at the same level where the current `4.0.3` tag exists currently then merge this PR. the moment this PR is merged github action will be triggered and `semantic-release` will start working.

The expected result is that it won't release any new version as all the types of commits I have committed were either `chore`, `ci`, or `build` and none of them was `feature` or `fix` type.

I have also put it on `dry-run` mode just for safety and forgot to mention that once everything is merged and working properly you will need to disable the `dry-run` mode. (I will comment on that after this)

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -10,20 +10,80 @@ on:
...
+        uses: actions/checkout@v2
+        with:
+        # latest nodejs LTS
+          node-version: 14.x
+      - run: npm ci
+      - name: Preparing a release for npm and github 🚀
+      # more about this action here: https://github.com/cycjimmy/semantic-release-action
+        uses: cycjimmy/semantic-release-action@v2
+        with:
+          dry_run: true
```
Comment (gimyboya) [inline .github/workflows/node.js.yml:83]: @sgtsquiggs @z3bi once you confirm that everything is working as expected you can remove the `dry_run` line for `semantic-release` to work properly

**Comment by gimyboya** [comment]
Looking forward to your feedback @z3bi @sgtsquiggs Let me know if you guys need anything on my side

**Comment by z3bi** [comment]
Lets go ahead with this PR. @gimyboya there are some merge conflicts that came up. If you can go ahead and resolve those I will create the tag we need.

**Comment by gimyboya** [comment]
@z3bi Conflict is waved. Thank you for taking the  time to review this PR

**Review by z3bi — APPROVED** [review]

**Comment by gimyboya** [comment]
@z3bi Thank you for mergin! 

Lastly don't forget to turn off the `dry-run` once you confirm that `semantic-release` is working as intended.

**Comment by z3bi** [comment]
@gimyboya yup we are just testing everything out right now. Thanks for this awesome PR!

**Pull request merged**